File: EcuCuadratica.py
import math
class EcuacionCuadratica:
    def __init__(self, x=0, y=0, z=0):
        self.x = x
        self.y = y
        self.z = z
    def setX(self, x):
        self.x = x
    def setY(self, y):
        self.y = y  
    def setZ(self, z):
        self.z = z 
    def get_discriminante(self):
        return self.y ** 2 - 4 * self.x * self.z
    
    def get_raiz(self):
        d = self.get_discriminante()
        if d >= 0:
            return (-self.y + math.sqrt(d)) / (2 * self.x)
        else:
            return None

    def get_raiz2(self):
        d = self.get_discriminante()
        if d >= 0:
            return (-self.y - math.sqrt(d)) / (2 * self.x)
        else:
            return None
    
    def __str__(self):
        d = self.get_discriminante()
        if d > 0:
            return f"tiene dos raices: {self.get_raiz()} y {self.get_raiz2()}"
        elif d == 0:
            return f"tiene una raiz: {self.get_raiz()}"
        else:
            return "raiz real"

eq1 = EcuacionCuadratica()
eq1.setX(1.0)
eq1.setY(3)
eq1.setZ(1)
print(eq1)

File: EcuCuadratica.java
import java.lang.Math;
class EcuacionCuadratica {
    private double x;
    private double y;
    private double z;

    public EcuacionCuadratica(double x, double y, double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
    public void setX(double x) {
        this.x = x;
    }
    public void setY(double y) {
        this.y = y;
    }
    public void setZ(double z) {
        this.z = z;
    }
    public double getDiscriminante() {
        return (y * y) - (4 * x * z);
    }
    public Double getRaiz() {
        double d = getDiscriminante();
        if (d >= 0) {
            return (-y + Math.sqrt(d)) / (2 * x);
        } else {
            return null;
        }
    }
    public Double getRaiz2() {
        double d = getDiscriminante();
        if (d >= 0) {
            return (-y - Math.sqrt(d)) / (2 * x);
        } else {
            return null;
        }
    }
    public String toString() {
        double d = getDiscriminante();
        if (d > 0) {
            return "Tiene dos raíces: " + getRaiz() + " y " + getRaiz2();
        } else if (d == 0) {
            return "Tiene una raíz: " + getRaiz();
        } else {
            return "Raíz real no disponible";
        }
    }
    public static void main(String[] args) {
        EcuacionCuadratica eq1 = new EcuacionCuadratica(1.0, 3, 1);
    System.out.println(eq1);
    }
}
